sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/Plugin.java

```diff
@@ -39,10 +40,19 @@
   private boolean supportedBySonarSource = false;
   private List<String> developers;
 
-  public Plugin(String key) {
+  private Plugin(String key) {
     super(key);
   }
 
+  public static Plugin factory(String key) throws IllegalArgumentException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the declaration of thrown exception 'java.lang.IllegalArgumentException' which is a runtime exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)